### PR TITLE
Site People: remove deleted users

### LINKED ISSUE #5675: People: Pull to refresh user list does not remove users who were deleted
### Expected behavior

While viewing the list of users on a site, pulling down should refresh the list of users:
- Add new users who were added to the site
- Remove users who were deleted from the site
- Update user details (display name, role, etc.)
### Actual behavior

Pulling down to refresh works to see new users added to the site and update existing user details. However, it does not remove users who were deleted from the site.
### Steps to reproduce the behavior
1. With a site that has multiple users, go to My Site > People in the app
2. Make sure you are viewing the list of users on the site in the app
3. In a web browser, go to https://wordpress.com/people/team
4. Select a user and remove them from the site
5. In the app, pull down to refresh

Expected: The deleted user is removed from the user list
Actual: The deleted user remains in the list, and:
- If you tap on that deleted user, you can still view the user details.
- Trying to change the user's role results in an error.
- If you try to remove the user again in the app you are returned to the user list and the deleted user is still there.

### PR DESCRIPTION
**Pull request by ScoutHarris**
Fixes #5675 

This fixes the issue of deleted users appearing in the People lists.

The issue was People were being added/updated in Core Data but never removed. This change removes all People from Core Data when the view is initially shown, and on pull to refresh. Core Data is repopulated as the People lists are loaded.

To test:
- Via the website (https://wordpress.com/people/team), invite a user.
- As that user, accept the invitation.
  - In the app, verify the user appears in Site > People.
- Via the website, remove that user.
  - In the app, refresh the People list.
  - Verify the user no longer appears.

Update release notes:
- [x] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

**Comment by ScoutHarris** [comment]
@frosty @aerych @stevebaranski - this isn't 100% complete (ex: there's some visual glitches when the view is first loaded), but before I potentially waste time doing that, I wanted to get your opinion on the problem solution itself.

Clearing and repopulating Core Data was the "best" solution I came up with to remove deleted users from Core Data - essentially starting fresh when People data refresh is needed.

If anyone has suggestions for a better/preferable solution, I'll take it. If this solution is acceptable, I'll continue with fixing the UI. 

Thanks!

File: WordPress/Classes/Services/PeopleService.swift [hunk 1]
```diff
@@ -211,6 +210,15 @@ struct PeopleService {
...
     }
 
+    /// Nukes all users from Core Data.
+    ///
+    func removeManagedPeople() {
+        let request = NSFetchRequest<NSFetchRequestResult>(entityName: "Person")
+        request.predicate = NSPredicate(format: "siteID = %@", NSNumber(value: siteID as Int))
+        let objects = (try? context.fetch(request) as! [NSManagedObject]) ?? []
+        objects.forEach({ context.delete($0) })
+    }
```
Comment (stevebaranski) [inline WordPress/Classes/Services/PeopleService.swift]: Small suggestion for lines 217-219:
```
request.predicate = NSPredicate(format: "siteID = %@", NSNumber(value: siteID))
if let objects = (try? context.fetch(request)) as? [NSManagedObject] {
    objects.forEach { context.delete($0) }
}
```

**Comment by stevebaranski** [comment]
@ScoutHarris thanks for sharing this with me. It's great to see a defect this old being tidied up!

In my testing, the "flush & fill" approach here behaves as desired. I'd be interested to see if our other reviewers have an alternate approach in mind.

File: WordPress/Classes/Services/PeopleService.swift (hunk 1 above)
Comment (ScoutHarris) [reply WordPress/Classes/Services/PeopleService.swift]: And done!

**Comment by ScoutHarris** [comment]
@stevebaranski  @frosty - I've added a loading view on initial view and refresh. If either of you could do one more review I'd appreciate it. Thanks!

**Comment by stevebaranski** [comment]
Hey @ScoutHarris - thanks for considering the feedback. The NRV looks great to me, so :shipit:!

**Comment by ScoutHarris** [comment]
Thanks @stevebaranski !

**Pull request merged**